### ISSUE TITLE
"TypeError: expected bytes, str found" in SQL Alchemy engine

### ISSUE THREAD

**Issue by ar-qun** [GoogleCloudPlatform/python-docs-samples#4854]
## In which file did you encounter the issue?

`site-packages/flask/app.py`

The actual error is happening in `/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 545`

### Did you change the file? If so, how?

`site-packages/sqlalchemy/engine/result.py`

```
            colname = rec[0]
            coltype = rec[1]

            if description_decoder:
                print(colname)
                colname = description_decoder(bytes(colname, 'ascii'))

            if translate_colname:
                colname, untranslated = translate_colname(colname)
```

added `bytes(..., 'ascii')`

## Describe the issue

When running the Cloud SQL PostgreSQL sqlalchemy I get this error:

```Traceback (most recent call last):
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 1945, in full_dispatch_request
    self.try_trigger_before_first_request_functions()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/flask/app.py", line 1993, in try_trigger_before_first_request_functions
    func()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/main.py", line 144, in create_tables
    with db.connect() as conn:
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2263, in connect
    return self._connection_cls(self, **kwargs)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 104, in __init__
    else engine.raw_connection()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2370, in raw_connection
    self.pool.unique_connection, _connection
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2336, in _wrap_pool_connect
    return fn()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 304, in unique_connection
    return _ConnectionFairy._checkout(self)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 778, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 495, in checkout
    rec = pool._do_get()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/impl.py", line 140, in _do_get
    self._dec_overflow()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/impl.py", line 137, in _do_get
    return self._create_connection()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 309, in _create_connection
    return _ConnectionRecord(self)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 440, in __init__
    self.__connect(first_connect_check=True)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 666, in __connect
    ).exec_once_unless_exception(self.connection, self)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/event/attr.py", line 314, in exec_once_unless_exception
    self._exec_once_impl(True, *args, **kw)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/event/attr.py", line 285, in _exec_once_impl
    self(*args, **kw)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/event/attr.py", line 322, in __call__
    fn(*args, **kw)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 1397, in go
    return once_fn(*arg, **kw)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/strategies.py", line 199, in first_connect
    dialect.initialize(c)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/pg8000.py", line 215, in initialize
    super(PGDialect_pg8000, self).initialize(connection)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/base.py", line 2611, in initialize
    super(PGDialect, self).initialize(connection)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 313, in initialize
    connection
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/base.py", line 2856, in _get_server_version_info
    v = connection.execute("select version()").scalar()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1003, in execute
    return self._execute_text(object_, multiparams, params)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1178, in _execute_text
    parameters,
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1514, in _handle_dbapi_exception
    util.raise_(exc_info[1], with_traceback=exc_info[2])
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1294, in _execute_context
    result = context._setup_crud_result_proxy()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 1243, in _setup_crud_result_proxy
    result = self.get_result_proxy()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 1218, in get_result_proxy
    return result.ResultProxy(self)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 775, in __init__
    self._init_metadata()
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 807, in _init_metadata
    self._metadata = ResultMetaData(self, cursor_description)
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 296, in __init__
    textual_ordered,
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 514, in _merge_cursor_description
    ) in raw_iterator
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 497, in <listcomp>
    (
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 621, in _merge_cols_by_none
    ) in self._colnames_from_description(context, cursor_description):
  File "/home/bugsbunny/python-docs-samples/cloud-sql/postgres/sqlalchemy/env/lib/python3.7/site-packages/sqlalchemy/engine/result.py", line 545, in _colnames_from_description
    colname = description_decoder(colname)
TypeError: expected bytes, str found
```

**Comment by ar-qun** [comment]
Related issue sqlalchemy/sqlalchemy#5645

Solution mentioned by [zzzeek](https://github.com/sqlalchemy/sqlalchemy/issues/5645#issuecomment-707879323):

```e = create_engine("postgresql+pg8000://...")
e.dialect.description_encoding = None```

**Comment by averikitsch** [comment]
I found that downgrading dependency pg8000 fixed this issue: `pip install pg8000==1.16.5`.
